.github/scripts/utils.py/update-crowdin-locales.py

```diff
@@ -0,0 +1,88 @@
+import os
+import logging
+import json
+
+from crowdin_api import CrowdinClient
+from fnmatch import fnmatch
+
+CROWDIN_PROJECT_ID = 51028
+CROWDIN_API_TOKEN = os.environ["CROWDIN_PAT"]
+DEFAULT_LOCALE = "en-US"
+
+crowdin_api = CrowdinClient(token=CROWDIN_API_TOKEN, project_id=CROWDIN_PROJECT_ID, page_size=500)
+
+logging.basicConfig()
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def get_dir_id(name: str) -> int:
+    crowdin_api.source_files.list_directories(filter=name)["data"][0]["data"]["id"]
```

Comment:
Is `list_directories` assured to always return a dictionary with an non-empty array with key `data`, which itself will contain objects that _always_ contain an object at key `data` which has a key `id`?

If the answer is "no" or "maybe" use a chain of `get` calls instead:

```Python
directories = crowdin_api.source_files.list_directories(filter=name).get("data", [])

if len(directories) > 0:
  id = directories[0].get("data", {}).get("id", None)
  if id is not None:
    return id
  else:
    raise Exception("No id field on directory data")
else:
  raise Exception("No crowdin source file directories found")
```